src/porepy/grids/coarsening.py

```diff
@@ -19,7 +19,7 @@
 
 
 def coarsen(
-    g: Union[pp.Grid, pp.MixedDimensionalGrid], method: str, **method_kwargs
+    grid: Union[pp.Grid, pp.MixedDimensionalGrid], method: str, **method_kwargs
```

Comment:
Why do you use grid here and g in obtain_interdim_mappings?

Reply:
My thinking was that this is either a pp.Grid or an mdg, while obtain_interdim_mappings is definitely a grid. There is no clear solution here, I'm open to applying a different resolution mechanism.